pkg/raft/term_cache_test.go

```diff
@@ -0,0 +1,396 @@
+			// Test case: [t6/11, t7/12, t8/13, t9/14, t9/15, t9/16]
+			want: entryID{term: 0, index: 10}.append(6, 7, 8, 9, 9, 9).LogSlice,
+		},
+		{
+			name:      "overwriting with new entries",
+			cacheSize: 4,
+			// Initial: [t5/10, t6/11, t7/12, t8/13]
+			init: entryID{term: 5, index: 10}.append(6, 7, 8).LogSlice,
+			// Append: [t5/10, t11/11, t12/12, t13/13]
+			app: entryID{term: 5, index: 10}.append(11, 12, 13).LogSlice,
+			// Test case: [t5/10, t11/11, t12/12, t13/13, t13/14, t13/15]
+			want: entryID{term: 0, index: 9}.append(5, 11, 12, 13, 13, 13).LogSlice,
+		},
+		{
+			name:      "repeated term appends won't modify the term cache",
+			cacheSize: 4,
+			// Initial: [t5/10, t6/11, t7/12, t8/13]
+			init: entryID{term: 5, index: 10}.append(6, 7, 8).LogSlice,
+			// Append: [t8/13, t8/14, t8/15, ... many repeating terms]
+			app: entryID{term: 8, index: 13}.append(8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8).LogSlice,
+			// Test case: [t6/11, t7/12, t8/13, t9/14, t9/15, t9/16]
+			want: entryID{term: 0, index: 9}.append(5, 6, 7, 8, 8, 8).LogSlice,
+		},
+		{
+			name:      "overwrite test case",
+			cacheSize: 4,
+			// Initial: [t5/10, t6/11, t7/12, t8/13]
+			init: entryID{term: 5, index: 10}.append(6, 7, 8).LogSlice,
+			// Append: [t5/10, t7/11, t8/12, ... t13/17, t13/18, ...]
+			app: entryID{term: 5, index: 10}.append(7, 8, 9, 10, 11, 12, 13, 13, 13, 13, 13, 13).LogSlice,
+			// Test case: [t10/14, t11/15, t12/16, t13/17, t13/18, ...]
+			want: entryID{term: 0, index: 13}.append(10, 11, 12, 13, 13, 13, 13, 13, 13, 13, 13).LogSlice,
+		},
+		{
+			name:      "extreme but possible, overwrite with LogSlice.prev and entry",
+			cacheSize: 4,
+			// Initial: [t5/10, t6/11, t7/12, t8/13]
+			init: entryID{term: 5, index: 10}.append(6, 7, 8).LogSlice,
+			// Append: [t4/9, t10/10]
+			app: entryID{term: 4, index: 9}.append(10).LogSlice,
+			// Test case: [t4/9, t10/10, t10/11, t10/12]
+			want: entryID{term: 0, index: 8}.append(4, 10, 10).LogSlice,
+		},
+		{
+			name:      "overwrite with just one entry",
+			cacheSize: 4,
+			// Initial: [t5/10, t6/11, t7/12, t8/13]
+			init: entryID{term: 5, index: 10}.append(6, 7, 8).LogSlice,
+			// Append: [t5/10, t10/11]
+			app: entryID{term: 5, index: 10}.append(10).LogSlice,
+			// Test case: [t5/10, t10/11, t10/12, t10/13]
+			want: entryID{term: 0, index: 9}.append(5, 10, 10, 10).LogSlice,
+		},
+		{
+			name:      "multiple gaps in between, test term",
+			cacheSize: 4,
+			// Initial: [t5/10, t6/11]
+			init: entryID{term: 5, index: 10}.append(6).LogSlice,
+			// Append: [t7/12, t7/13, t7/14, t9/15, t9/16, t9/17, t10/18, t10/19, t10/20]
+			app: entryID{term: 6, index: 11}.append(7, 7, 7, 9, 9, 9, 10, 10, 10).LogSlice,
+			// Test case: [t6/11, t7/12, t7/13, t7/14, t9/15, t9/16, t9/17, t10/18, t10/19, t10/20]
+			want: entryID{term: 6, index: 10}.append(6, 7, 7, 7, 9, 9, 9, 10, 10, 10).LogSlice,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
nit: the `tc.name` is verbose, and `testing` package will remove all punctuation from them. They will render as quite lengthy and unreadable test names, also not entirely searchable because of the punctuation change. You can see that by running `dev test` with the `--verbose` flag.

Suggest removing `tc.name`, and leaving `t.Run("", func ...)` here. Go `testing` package automatically inserts a subtest index to the name, so if any of these tests fails it is pretty easy to locate. The descriptions of each test can be just comments.